cdap-app-fabric/src/test/java/co/cask/cdap/stream/store/MDSStreamMetaStoreTest.java

```diff
@@ -0,0 +1,111 @@
+import co.cask.cdap.data.stream.service.StreamMetaStore;
+import co.cask.cdap.data2.datafabric.dataset.service.DatasetService;
+import co.cask.cdap.data2.dataset2.DatasetFramework;
+import co.cask.cdap.explore.guice.ExploreClientModule;
+import co.cask.cdap.gateway.auth.AuthModule;
+import co.cask.cdap.internal.app.store.DefaultStore;
+import co.cask.cdap.proto.Id;
+import co.cask.cdap.proto.NamespaceMeta;
+import co.cask.tephra.TransactionManager;
+import co.cask.tephra.TransactionSystemClient;
+import com.google.inject.AbstractModule;
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+import com.google.inject.Scopes;
+import com.google.inject.Singleton;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.twill.filesystem.LocationFactory;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+
+/**
+ * Test the {@link MDSStreamMetaStore}.
+ */
+public class MDSStreamMetaStoreTest extends StreamMetaStoreTestBase {
+
+  private static StreamMetaStore streamMetaStore;
+  private static DatasetService datasetService;
+  private static TransactionManager transactionManager;
+  private static Store store;
+
+  @BeforeClass
+  public static void init() throws Exception {
+    Injector injector = Guice.createInjector(
+      new ConfigModule(CConfiguration.create(), new Configuration()),
+      new DataSetServiceModules().getInMemoryModule(),
+      new DataSetsModules().getLocalModule(),
+      new DataFabricModules().getInMemoryModules(),
+      new ExploreClientModule(),
+      new AuthModule(),
+      new DiscoveryRuntimeModule().getInMemoryModules(),
+      new LocationRuntimeModule().getInMemoryModules(),
+      new AbstractModule() {
+        @Override
+        protected void configure() {
+          bind(StreamMetaStore.class).to(MDSStreamMetaStore.class).in(Scopes.SINGLETON);
+          bind(MetricsCollectionService.class).to(NoOpMetricsCollectionService.class).in(Singleton.class);
+        }
+      }
+    );
+
+    streamMetaStore = injector.getInstance(StreamMetaStore.class);
+    transactionManager = injector.getInstance(TransactionManager.class);
+    transactionManager.startAndWait();
+    datasetService = injector.getInstance(DatasetService.class);
+    datasetService.startAndWait();
+    store = new DefaultStore(injector.getInstance(CConfiguration.class), injector.getInstance(LocationFactory.class),
+                             injector.getInstance(TransactionSystemClient.class),
+                             injector.getInstance(DatasetFramework.class));
+  }
+
+  @AfterClass
+  public static void destroy() throws Exception {
+    datasetService.stopAndWait();
+    transactionManager.stopAndWait();
+  }
+
+  @Override
+  protected StreamMetaStore getStreamMetaStore() {
+    return streamMetaStore;
+  }
+
+  @Override
+  protected void createNamespace(String namespaceId) {
+    store.createNamespace(new NamespaceMeta.Builder().setId(namespaceId).build());
```

Comment:
I see your point about having a ``NamespaceService`` now. Would be useful in such situations, even though I think this particular one will go away after we namespace streams. Will do it soon. :)